### ISSUE TITLE
Enhanced edition not working with Windows XP

### ISSUE THREAD

**Issue by yukinsaknos** [elishacloud/Silent-Hill-2-Enhancements#146]
Hello,
I am getting a No entry point after i did put the new executable, D3D8 files and Enhanced Edition Essential Files.

On my win10 computer i do not have any issue but on the Windows XP the game will not launch.

Thanks

**Comment by AeroWidescreen** [comment]
Based on the following information, Windows XP is not supported and will probably never be supported:

> This project is written in C++ using Microsoft Visual Studio Community 2017.
> 
> The project uses the Windows 10 SDK and WDK. The exact version required can be seen in the project properties in Visual Studio.

[Windows 10 SDK Support](https://developer.microsoft.com/en-us/windows/downloads/windows-10-sdk):
> - Windows 10 version 1507 or higher
> - Windows Server 2016: Standard and Datacenter
> - Windows 8.1
> - Windows Server 2012 R2
> - Windows 7 SP1

**Comment by elishacloud** [comment]
@AeroWidescreen is correct.  This project was compiled with Microsoft Visual Studio Community 2017 and does not natively support Windows XP.  There is a way to compile the project for Windows XP, but that would take some work to support and that option will disappear when we move to Visual Studio 2019.  At this point there are no plans to support Windows XP for this project.

**Comment by Psycho-A** [comment]
I think it's not really correct to drop WinXP support for game that released in 2002 and still supports very old PC specs despite of all enhancements. Old-school gamers may like to play the enhanced version too, as it aims not only enhancements, but some critical compatibility fixes. Also, what's need to move the project into VS 2019? Without getting any profits it's not makes sense.

**Comment by elishacloud** [comment]
@Psycho-A, We didn't exactly drop support for Windows XP.  Microsoft actually dropped support for it.  My dev machine uses Microsoft Visual Studio Community 2017 for all my projects, which does not natively support Windows XP.  I collaborate with others on different GitHub projects and I would like to do all things in the same version of VS.  Some of the projects I collaborate in, like d3d8to9, has already moved to VS 2019.

I primarily focus on getting old games running on newer OS's.  So typically my projects don't need to be run on Windows XP, as the games I work on already run on Windows XP.  However, there are still a good number of them that don't run on Windows 10.  That's where I come in.  Anyways, I don't even have a Windows XP machine to test it on and I don't really want to spend the time to build one.

Maybe if I have a few minutes I will see if I can get it to compile for Windows XP, but for now I am booked solid with other things!

**Comment by Psycho-A** [comment]
> Maybe if I have a few minutes I will see if I can get it to compile for Windows XP, but for now I am booked solid with other things!

If you can, it would be great, and I even may help you with testing your result since I have 2004-yr notebook with Windows XP onboard.

**Comment by elishacloud** [comment]
Ok, I was able to target this project for Windows XP and recompile it.  This should work in Windows XP.

Try this one and let me know if it works or not: [d3d8.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/3571964/d3d8.zip)

**Comment by Psycho-A** [comment]
@elishacloud
Finally I was able to reach home and machine with Win XP. What can I say.. it works! But only if we remove IndirectSound's dsound.dll from SH2 folder. It seems it not fully compatible with XP, or it's just my system's/drivers issue. Here's dsound.log anyway...


```
[[============================================================]]
  This file is created every time that IndirectSound is loaded
[[============================================================]]

IndirectSound was loaded statically at process startup
The IndirectSound library being loaded is located at X:\Games\Classics Gamepad\Silent Hill (Series)\Silent Hill 2. Director's Cut\App\DSOUND.dll
	Version -- 0.17
The process loading IndirectSound is located at X:\Games\Classics Gamepad\Silent Hill (Series)\Silent Hill 2. Director's Cut\App\sh2pc.exe
	File version info doesn't exist
Initializing User Settings
Loaded Local User Settings file: X:\Games\Classics Gamepad\Silent Hill (Series)\Silent Hill 2. Director's Cut\App\dsound.ini
hardwareBufferCount: 128
	128 hardware-accelerated buffers will be emulated
eax1: false
	EAX 1.0 support will be disabled
eax2: false
	EAX 2.0 support will be disabled
surroundPanning: "constantPower"
	3D sounds will be positioned in speakers using constant power (-3 dB) panning
maxFrequencyRatio: 2
	Sounds can be played up to 2 times faster than their original frequency
	(pitch can be raised 2 times higher)
monoSourceBehavior: "center"
	2D sounds with mono sources will play out of the front center speaker if it exists
	and the front left and right speakers otherwise
disabled3dBehavior: "2D"
	3D sounds with disabled processing will play out of the same speakers that they would if they were 2D
earlyStopFadeOutTime: 0.002
	Sounds that are stopped before they are finished will fade out over 0.002 seconds
nearDistanceToStartNormalizing: 0.083
	3D sounds will gradually be normalized in all speakers when they are located 0.083 world units or closer to the listener
speedOfSound: 343.5
	Doppler shift will be calculated with a reference speed of sound of 343.5 meters per second
milesCompatibility: false
	Miles Sound System compatibility mode will be disabled
The requested audio device {BD6DD71A-3DEB-11D1-B171-00C04FC20002} was not found!
	Please contact indirectsound@gmail.com
A mastering voice can't be created without a valid device index
The new instance of IDirectSound didn't successfully initialize
IXAudio2::GetDeviceDetails() failed after finding the device index for a new mastering voice!
	Please contact indirectsound@gmail.com
The new instance of IDirectSound8 didn't successfully initialize
```

**Comment by elishacloud** [comment]
We did not write IndirectSound.  However, it was written in Visual Studio 2010 so it should work fine with Windows XP.  In order for it to work you need the following two items installed:

* [Microsoft Visual C++ 2010 Redistributable Package (x86)](https://www.microsoft.com/en-us/download/details.aspx?id=5555)
* [DirectX 9 SDK](https://www.microsoft.com/download/details.aspx?id=35)

For more details you can refer to this site: [https://www.indirectsound.com/](https://www.indirectsound.com/)

**Comment by Psycho-A** [comment]
I learned more about IndirectSound and figured out that's for modern OSes only, because it emulates hardware functions that were loose since Windows 7 and later. No matter if it's compiled with VS2010. So, for EE's Windows XP installation we just need to exclude IndirectSound component, besides game support most of it's features natively on Windows XP.
>> IndirectSound emulates audio hardware acceleration on modern Windows operating systems. This enables older games to have 3D positional audio (i.e. surround sound played out of rear and side speakers) like they were intended to when they were originally released.
Where Do I Get It?

Download the latest IndirectSound: Version 0.20, released 2019-09-21. All currently-known issues are listed here and previous versions are available to download here.
Why Would I Want It?

You should consider using IndirectSound if the following apply:

    You have more than two computer speakers (e.g. rear/surround, 5.1, 7.1, etc.)
    You have a Windows operating system newer than XP (i.e. Vista or later)
    You still play older computer games
        A list of games whose 3D positional surround sound has been reported to be restored by IndirectSound are listed here
        For examples of games that might benefit from using IndirectSound see this list (towards the bottom of the page) of games that are officially supported by Creative ALchemy and this unofficial list of games which users say ALchemy works with (look for "DirectSound3D" in the API column)
    You don't already have an existing solution that re-enables audio hardware acceleration for your sound card
        For example, if you have a Sound Blaster and use Creative ALchemy then you do not need IndirectSound 

If you meet the above criteria then IndirectSound can restore 3D positional surround sound that is otherwise missing in some games when played on modern versions of Windows.

**Comment by Psycho-A** [comment]
So, will this merged into a main EE release, if it's already not?
If there are no any functionality looses, I don't see the reason not doing that...

**Comment by Polymega** [comment]
> So, will this merged into a main EE release, if it's already not?

Do you mean dropping IndirectSound from the installation method? We will keep IndirectSound as part of the process, as a main goal of this project is to get the game running on modern hardware/displays, along with the improvements we're doing. 

If someone else is using Windows XP and runs into an issue, we can inform them to drop IndirectSound at that time.

**Comment by Psycho-A** [comment]
No, I'd suggest just to write on your site in "6. Install IndirectSound" that if you using WinXP, you shouldn't install IndirectSound. That would be pretty enough.

**Comment by mirh** [comment]
Even if the developers use VS2019, if you there's no special C++20 feature in want, I suppose v141_xp shouldn't have any downside for the years to come. 

Said this, putting aside that I'd expect a 2004 computer with XP to have all the right drivers for directsound to work properly... [DSOAL](https://github.com/kcat/dsoal) is also a thing.

**Comment by elishacloud** [comment]
> I suppose v141_xp shouldn't have any downside for the years to come.

Good point.  We can continue to use "v141_xp" even with Visual Studio 2019, if we [install the component](https://docs.microsoft.com/en-us/cpp/build/configuring-programs-for-windows-xp?view=vs-2019#install-the-windows-xp-platform-toolset).

As far as [DSOAL](https://github.com/kcat/dsoal), we want to use it in this project, but there is currently some specific code in IndirectSound that we require.  The author of IndirectSound added a feature to allow us to remove the pop/click sound at the end of sfx files that get clipped.  I am hoping to add the feature to this project natively, but until I do that we will continue to use IndirectSound.

**Comment by mirh** [comment]
That landed in 21072fea74b22ced9cfc043e2115c95667d6c992 I guess, and #278 should have seen the switch to dsoal. 
(also https://github.com/elishacloud/Silent-Hill-2-Enhancements/issues/423#issuecomment-835536453 argued that this was already fixed)

The only thing to keep into consideration is https://github.com/kcat/dsoal/issues/22

**Comment by elishacloud** [comment]
This issue is resolved in the latest release.  Note: DSOAL does not work on XP, but should not be needed on XP either.
